mapserver/django/django_project/mapserver/api/wms.py

```diff
@@ -0,0 +1,79 @@
+__author__ = 'Irwan Fathurrahman <irwan@kartoza.com>'
+__date__ = '09/06/20'
+
+import json
+import os
+import requests
+from urllib.parse import urlparse, parse_qs, urlencode
+from django.conf import settings
+from django.http import (
+    HttpResponse, HttpResponseForbidden, HttpResponseBadRequest,
+    HttpResponseServerError)
+
+
+def get_sld_style(layer_name):
+    """ return sld style in json
+    :param layer_name: the layer name that needs to be checked
+    :return:
+    """
+    fixture_file = os.path.join(settings.FIXTURES, 'data', 'layer_style.json')
+    with open(fixture_file, 'r') as _file:
+        sld = json.load(_file).get(layer_name, None)
+        if sld:
+            return os.path.join(settings.MAPSERVER_PUBLIC_SLD_URL, sld)
+    return None
+
+
+def redirect_mapserver(request, mapserver_url):
+    """ Return updated url for redirect to mapserver
+    :param request:
+    :return: params
+    :rtype: str
+    """
+    try:
+        if request.method == 'GET':
+            # Parse querystring
+            parse_result = urlparse(request.build_absolute_uri())
+            query_params = parse_qs(parse_result.query)
+            # make dictionary key all lowercase
+            params = dict((k.lower(), v) for k, v in query_params.items())
+            if 'getmap' == params.get('request', [''])[0].lower():
+                layers = params.get('layers')
+                layers = [l.split(':')[-1] for l in layers]
```

Comment:
is it the case for not just kartoza? i thought it is just kartoza:

Reply:
well, we have no case like that, but the reason why I changed it is because previously you strip it from the whole query string instead of just for the required parameter.